### PR TITLE
Remove `cluster.routing.allocation.disk.watermark.enable_for_single_data_node` setting

### PR DESCRIPTION
**Pull request by arteam**
Prior to 7.8, whenever a cluster had only a single data node, the watermarks would not be respected. This was incompatible with how storage based autoscaling works and in order to change this in 7.8+ in a backwards compatible way, we had to introduce the `cluster.routing.allocation.disk.watermark.enable_for_single_data_node node` setting. 

The setting was deprecated in https://github.com/elastic/elasticsearch/pull/73733 (7.14), and was made to accept only `true` in https://github.com/elastic/elasticsearch/pull/73737 (8.0).

### CODE REVIEW

**Comment by arteam** [comment]
> We haven't yet decided that we want to do this in v9, let's hold off here until that's settled.

@DaveCTurner Yes, I'm aware that the breaking change hasn't been approved yet, that's why I created this PR as a draft and added a `WIP` label.

**Review by henningandersen — COMMENTED** [review]
Looks good, though I think we should add a breaking change changelog entry to this PR. I thought that was half-automatic, but perhaps not from draft?

**Comment by arteam** [comment]
@elasticmachine update bramch

**Comment by arteam** [comment]
@elasticmachine update branch

**Comment by arteam** [comment]
Hi @henningandersen! I've made the PR ready to review and added the breaking changelog entry.

File: docs/changelog/114207.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+type: breaking
+issues: []
+breaking:
+  title: Remove `cluster.routing.allocation.disk.watermark.enable_for_single_data_node` setting
+  area: 'Cluster and node setting'
+  details: >-
+    Prior to 7.8, whenever a cluster had only a single data node, the watermarks would not be respected.
+    This was incompatible with how storage based autoscaling works and in order to change this in 7.8+ in a backwards
+    compatible way, we had to introduce the `cluster.routing.allocation.disk.watermark.enable_for_single_data_node`
+    node setting. The setting was deprecated in 7.14 and was made to accept only true in 8.0
```
Comment (henningandersen) [inline docs/changelog/114207.yaml:13]: I think there is no need to repeat why we did this, that is long gone history.

```suggestion
  details: >-
    Prior to 7.8, whenever a cluster had only a single data node, the watermarks would not be respected.
    In order to change this in 7.8+ in a backwards compatible way, we introduced the 
    `cluster.routing.allocation.disk.watermark.enable_for_single_data_node`
    node setting. The setting was deprecated in 7.14 and was made to accept only true in 8.0
```

**Review by henningandersen — APPROVED** [review]
LGTM (with one breaking change documentation change).

**Comment by leemthompo** [comment]
@arteam is this PR relevant to the serverless changelog? [FYI this question is based on [9.0 breaking changes](https://staging-website.elastic.co/docs/release-notes/elasticsearch/breaking-changes)]

**Comment by arteam** [comment]
@leemthompo No, this change doesn't have any effect on ES Serverless

**Pull request merged**